docs/EDGEVIEW-CONTAINER-API.md

```diff
@@ -23,3 +23,69 @@ For security reasons, the Edge-View container has all the volumes mounted in 're
 ## Logging to Event
 
 Edge-View logging is similar to other containers. For any user command received by Edge-View, it will log the client endpoint(IP address/port), the command and its parameters. The log entry will also be tagged with object-type of `log-to-event`, and the controller can optionally process those log entries and generate them as device events or alerts.
+
+## Edge-View Client Authentication
+
+To address limitations in controlling and tracking user access to Edge-View commands, the authentication mechanism to Edge-View aim to strengthen user authentication, enforce stricter access controls, and improve user tracking.
+
+### Client Authentication Features
+
+1. **Scope**
+   Stricter authentication methods are implemented to ensure that commands originate from authorized sources. These sources include the Controller webpage session or authenticated remote users. JWT token is extended to support types of client authentication.
+
+2. **Edgeview Policy Addition**
+   A new optional "Controller Auth Only" item will be added to the controller Edge-View policy. This policy, definable at the project level, enforces that Edgeview commands must originate from the Controller.
+
+3. **Client Authentication (Controller Cert Signed Type)**
+   Commands issued from the Edge-View UI will be signed by the Zedcloud/Controller's private signing key, if the policy requires it. The device will verify these commands using the corresponding public certificate from the controller.
+
+4. **Remote User Authentication (SSH Key Pairs Type)**
+   Remote users can authenticate using SSH RSA keys, providing an additional secure access method for Edge-View. These public keys will be installed via the Edge-node ConfigItem "edgeview.authen.publickey". If there are changes to the "edgeview.authen.publickey" ConfigItem, any active Edge-View sessions on the Edge-node will be terminated. Users will need to re-enable the Edge-View session to apply the updated authentication settings. This ensures that only authorized keys are used for remote access.
```

Comment:
Is this limited to SSH RSA keys? Or can I use any SSH key such as DSA? 
If it isn't only about RSA then drop "RSA" from the sentence.

Reply:
For now with this patch, it is only for SSH RSA Key. I have not tried DSA, but I have tried 'ECDSA' type, with the golang for signing/verifying, I could not make it work. So for this patch, it is only for SSH RSA.

Reply:
So the ECC code is for the signing/verifying for the controller case?

If a user has a ssh dsa key what error do they see? It probably makes sense to document that since a ssh user might not know which key(s) they have.

Reply:
i'll add to the error message ', only SSH RSA keys are supported' to the above to make it clear.

Reply:
yes, the ECC code in the patch is for controller certs.

Reply:
Yes, please add that.

Reply:
updated the error message.

Reply:
@eriknordmark I have made the SSH key working with the ECC types also, updated the patch. I removed the explicit document reference to RSA, and have the error message to spell out both RSA and ECC are supported types.